src/node/rpc/member_frontend.h

```diff
@@ -387,6 +394,68 @@ namespace ccf
         caller_id, {cose_sign1.begin(), cose_sign1.end()});
     }
 
+    ProposalSubmissionStatus is_proposal_submission_acceptable(
+      kv::Tx& tx,
+      const std::string& created_at,
+      const std::vector<uint8_t>& request_digest,
+      const ccf::ProposalId& proposal_id,
+      ccf::ProposalId& colliding_proposal_id,
+      std::string& min_created_at)
+    {
+      auto cose_recent_proposals = tx.rw(network.cose_recent_proposals);
+      auto key = fmt::format("{}:{}", created_at, ds::to_hex(request_digest));
+
+      std::vector<std::string> replay_keys;
+      cose_recent_proposals->foreach_key(
+        [&replay_keys](const std::string& replay_key) {
+          replay_keys.push_back(replay_key);
+          return true;
+        });
+
+      std::sort(replay_keys.begin(), replay_keys.end());
```

Comment:
We should write a comment somewhere (and/or raise a GitHub issue) to use sorted tables for this once we have support for them?

Reply:
@jumaffre appended as https://github.com/microsoft/CCF/issues/3088#issuecomment-1400708471